### PR TITLE
Add "explicitFileExtensions" option for nodenext module resolution

### LINKED ISSUE #13: Support projects using nodenext module resolution
When using nodenext/node16 module resolution, source imports must include the .js extension in the path, causing the following error in the generated schemas when `__nullable__` is used:

> `Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './__nullable__.js'?`

Perhaps a config option to include the file extensions should be added?

### PR DESCRIPTION
**Pull request by Dan-Shields**
Closes #13

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ const configSchema = Type.Object(
      */
     nullableName: Type.String({ default: "__nullable__" }),
     /**
-     * Wethter to allow recursion in the generated schemes (enabling reduces code size)
+     * Wether to allow recursion in the generated schemes (enabling reduces code size)
```
Comment (m1212e) [inline src/config.ts:43]: Nice catch, thanks!

File: src/config.ts [hunk 2]
```diff
@@ -63,6 +63,10 @@ const configSchema = Type.Object(
      * E.g. Date will be of Type String when enabled.
      */
     useJsonTypes: Type.Boolean({ default: false }),
+    /**
+     * Wether to add ".js" extensions to src file imports to support nodenext module resolution
+     */
+    explicitFileExtensions: Type.Boolean({ default: false }),
```
Comment (m1212e) [inline src/config.ts:69]: Is there a possible scenario where this should be configurable? like I now want this to be '.mjs' but for my other project I need this to be '.ts' or just '.js'? Is this a thing or will it just never happen?

File: src/generators/wrappers/nullable.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ export const ${getConfig().nullableName} = <T extends TSchema>(schema: T) => ${
 export function nullableImport() {
   return `import { ${getConfig().nullableName} } from "./${
     getConfig().nullableName
-  }"\n`;
+  }${getConfig().explicitFileExtensions ? ".js" : ""}"\n`;
```
Comment (m1212e) [inline src/generators/wrappers/nullable.ts:15]: This should work if we don't need a customizeable ending. 

Is this change enough? I think we never actually import any modules somewhere else, besides the barrel file. Could you take a look at the barrel file and check real quick if the imports do work in your setup? Maybe that thing needs the adjusted imports too!

**Review by m1212e — CHANGES_REQUESTED** [review]

File: src/config.ts (hunk 1 above)
Comment (Dan-Shields) [reply src/config.ts:43]: Still spelt it incorrectly, oops... Should be "whether"

File: src/generators/wrappers/nullable.ts (hunk 3 above)
Comment (Dan-Shields) [reply src/generators/wrappers/nullable.ts]: Oh good catch! I don't use the barrel file so didn't spot that.

File: src/config.ts (hunk 2 above)
Comment (Dan-Shields) [reply src/config.ts:69]: Unsure if its important but might as well make it configurable. I've pushed an update for this

**Review by m1212e — APPROVED** [review]
Thanks!

**Pull request merged**